src/io/flutter/vmService/VmServiceWrapper.java

```diff
@@ -343,6 +345,52 @@ else if (errorResponses.size() > 0) {
         private void checkDone() {
           if (counter.decrementAndGet() == 0 && onFinished != null) {
             onFinished.run();
+
+            myVmService.getIsolate(isolateId, new GetIsolateConsumer() {
+              @Override
+              public void received(Isolate response) {
+                final Set<String> libraryUris = new HashSet<>();
+                for (LibraryRef library : response.getLibraries()) {
+                  libraryUris.add(library.getUri());
+                }
+
+                final ElementList<Breakpoint> breakpoints = response.getBreakpoints();
+                if (breakpoints.isEmpty()) {
+                  return;
+                }
+
+                final Analytics analytics = FlutterInitializer.getAnalytics();
+                final String category = "breakpoint";
+                boolean scriptFound = false;
+                for (Breakpoint breakpoint : breakpoints) {
+                  Object location = breakpoint.getLocation();
+                  if (location instanceof UnresolvedSourceLocation) {
+                    final ScriptRef script = ((UnresolvedSourceLocation)location).getScript();
+                    if (script != null) {
+                      if (libraryUris.contains(script.getUri())) {
```

Comment:
comment on this logic. Why do we expect some unresolved source locations to match the URIs of scripts we have found?

Reply:
It seems that locations are only considered "resolved" after a breakpoint has been triggered (although I'm not sure exactly when this happens). At the time that mapping is set up during app startup, all of the breakpoints are unresolved.